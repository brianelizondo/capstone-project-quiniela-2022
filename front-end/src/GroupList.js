import React, { useState, useEffect } from 'react';
import Loading from './Loading';
// import APIFootball api
import APIFootball from "./api-football";

import GroupStandings from './GroupStandings';

function GroupList(){
    // initial state
    const [loading, setLoading] = useState(false);
    const groups = ["A", "B", "C", "D", "E", "F", "G", "H"];
    const [groupsStandings, setGroupsStandings] = useState([]);

    useEffect(() => {
        setLoading(true);
        async function getGroupsStandings() {
            const resp = await APIFootball.getGroupsStandings();
            setGroupsStandings(resp);
        }
        getGroupsStandings();
        setLoading(false);
    }, []);

    if(loading){
        return <Loading />;
    }

    return (
        <div className="GroupList col-md-8 offset-md-2">
            <h1>Groups in FIFA World Cup 2022</h1>
            <p>Standings for each group</p>
            
            <div className="GroupList-details">
                { groups.map(group => (<GroupStandings key={group} group={group} standings={ groupsStandings.filter(team => team.group == group) } detailsButton={true} />))}
            </div>
        </div>
    );
}

export default GroupList;