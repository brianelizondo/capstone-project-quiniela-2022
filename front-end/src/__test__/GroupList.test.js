import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { MemoryRouter } from "react-router-dom";

import GroupList from '../GroupList';

// Smoke Test
describe("Smoke Test", () => {
    it("renders 'GroupList' component without crashing", () => {
        render(
            <MemoryRouter initialEntries={["/groups"]}>
                <GroupList />
            </MemoryRouter>
        );
        const textFound = screen.getByText("Standings for each group");
        expect(textFound).toBeInTheDocument();
    });
});

// Snapshot Tests
describe("Snapshot Test", () => {
    it("'GroupList' matches snapshot", () => {
        const {asFragment} = render(
            <MemoryRouter initialEntries={["/groups"]}>
                <GroupList />
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});

describe("Additional Test", () => {
    it("works when you click on a groups details button go to the group details", async () => {
        render(
            <MemoryRouter initialEntries={["/groups"]}>
                <GroupList />
            </MemoryRouter>
        );
        const user = userEvent.setup()

        waitFor(() => { expect(screen.getByText("Groups in FIFA World Cup 2022")).toBeInTheDocument() });
        waitFor(async () => { await user.click(screen.getAllByRole("button")) });
        waitFor(() => { expect(screen.getByText("Details for group")).toBeInTheDocument() });
    });
});