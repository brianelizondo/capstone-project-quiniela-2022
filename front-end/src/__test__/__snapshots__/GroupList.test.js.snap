// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Test 'GroupList' matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="GroupList col-md-8 offset-md-2"
  >
    <h1>
      Groups in FIFA World Cup 2022
    </h1>
    <p>
      Standings for each group
    </p>
    <div
      class="GroupList-details"
    >
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group A
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupA_button"
            type="button"
          >
            Group A Details
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group B
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupB_button"
            type="button"
          >
            Group B Details
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group C
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupC_button"
            type="button"
          >
            Group C Details
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group D
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupD_button"
            type="button"
          >
            Group D Details
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group E
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupE_button"
            type="button"
          >
            Group E Details
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group F
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupF_button"
            type="button"
          >
            Group F Details
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group G
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupG_button"
            type="button"
          >
            Group G Details
          </button>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="card-body"
        >
          <div
            class="card-title h5"
          >
            Group H
          </div>
          <table
            class="table table-sm table-striped"
          >
            <thead>
              <tr>
                <th>
                  Team
                </th>
                <th>
                  Played
                </th>
                <th>
                  Won
                </th>
                <th>
                  Draws
                </th>
                <th>
                  Lost
                </th>
                <th>
                  For
                </th>
                <th>
                  Against
                </th>
                <th>
                  Goal Diff
                </th>
                <th>
                  Points
                </th>
              </tr>
            </thead>
            <tbody />
          </table>
          <button
            class="btn btn-info btn-sm"
            id="groupH_button"
            type="button"
          >
            Group H Details
          </button>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
